From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 18 Jul 2026 13:56:38 +0200
Subject: clk: sunxi-ng: a64: Don't skip the 24MHz oscillator during switch

Keep 24 MHz oscillator switch patch until bare-metal terster verifies
safety of new NKMP factor selection algorithm.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/clk/sunxi-ng/ccu-sun50i-a64.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/clk/sunxi-ng/ccu-sun50i-a64.c b/drivers/clk/sunxi-ng/ccu-sun50i-a64.c
index 111111111111..222222222222 100644
--- a/drivers/clk/sunxi-ng/ccu-sun50i-a64.c
+++ b/drivers/clk/sunxi-ng/ccu-sun50i-a64.c
@@ -957,7 +957,6 @@ static struct ccu_pll_nb sun50i_a64_pll_cpu_nb = {
 	/* copy from pll_cpux_clk */
 	.enable	= BIT(31),
 	.lock	= BIT(28),
-	.once	= 1,
 };
 
 static struct ccu_mux_nb sun50i_a64_cpu_nb = {
@@ -965,7 +964,6 @@ static struct ccu_mux_nb sun50i_a64_cpu_nb = {
 	.cm		= &cpux_clk.mux,
 	.delay_us	= 1, /* > 8 clock cycles at 24 MHz */
 	.bypass_index	= 1, /* index of 24 MHz oscillator */
-	.once		= 1,
 };
 
 static int sun50i_a64_ccu_probe(struct platform_device *pdev)
-- 
Armbian

